### PR TITLE
Stop using CronJob for log rotation

### PR DESCRIPTION
**Pull request by d-kuro**
refs: https://github.com/cybozu-go/moco/issues/173

> ⚠️ This pull request depends on https://github.com/cybozu-go/moco-agent/pull/10
>
> Two PR changes are required for it to work properly:
>
> * https://github.com/cybozu-go/moco-agent/pull/10
> * https://github.com/cybozu-go/moco/pull/190

## Overview

Stop using CronJob for log rotation.
Instead, configure the agent using the log-rotation-schedule option added in https://github.com/cybozu-go/moco-agent/pull/10.

## Tests

E2E tests will not work properly without a Container Image with modified moco-agent code.

This is the result of running it on my local machine:

```console
$ pwd # moco-agent repo
/path/to/dir/github.com/cybozu-go/moco-agent

$ git branch # refs: https://github.com/cybozu-go/moco-agent/pull/10
* d-kuro/remove-rotate-api
  main

$ docker build . -t moco-agent:dev # Create a container image with the modified code
$ kind load docker-image moco-agent:dev --name moco-e2e # Load the kind cluster
```

```console
$ pwd # moco repo
/path/to/dir/github.com/cybozu-go/moco

$ git diff # add the binary-copy-container-image options for controller
diff --git a/e2e/manifests/moco_controller.yaml b/e2e/manifests/moco_controller.yaml
index 5ed7b0e..e1a6161 100644
--- a/e2e/manifests/moco_controller.yaml
+++ b/e2e/manifests/moco_controller.yaml
@@ -9,3 +9,6 @@ spec:
       containers:
       - name: manager
         imagePullPolicy: Never
+        args:
+          - "--binary-copy-container-image"
+          - "moco-agent:dev"

$ BOOTSTRAP=1 make test
...
=== RUN   TestE2E
Running Suite: kind test
========================
Random Seed: 1613282499
Will run 3 of 3 specs
...
Ran 3 of 3 Specs in 3.435 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestE2E (3.44s)
PASS
ok  	github.com/cybozu-go/moco/e2e	3.934s

$ kubectl get po -n e2e-test # working fine MySQL cluster
NAME                  READY   STATUS    RESTARTS   AGE
moco-mysqlcluster-0   4/4     Running   0          35m
moco-mysqlcluster-1   4/4     Running   0          35m
moco-mysqlcluster-2   4/4     Running   0          35m
```

## Review Poinsts

I'm wondering what to do with `MySQLCluster.spec.logRotationSecurityContext`.

### Option 1

Remove `MySQLCluster.spec.logRotationSecurityContext`.

### Option 2

Rename `MySQLCluster.spec.logRotationSecurityContext` -> `MySQLCluster.spec.agentSecurityContext`.

Which would be better?

### CODE REVIEW

**Comment by d-kuro** [comment]
> I'm wondering what to do with MySQLCluster.spec.logRotationSecurityContext.

After consulting with @kfyharukz, it was decided to remove the `logRotationSecurityContext` field.

**Review by masa213f — COMMENTED** [review]
Could you update documents and a test?

1. Remove the `logRotationSecurityContext` field from the following docs.
   - docs/crd_mysql_cluster.md
   - docs/example_mysql_cluster.md

1. Remove `"cornjobs"`, `"jobs"` from the following test.
   - e2e/gc_test.go

**Comment by d-kuro** [comment]
@masa213f 

Thanks for review 🙏 
I fix and pushed:
https://github.com/cybozu-go/moco/pull/190/commits/bcb134341997d47cdbd7076182c00099a0e36a70

```
•
Ran 3 of 3 Specs in 3.493 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestE2E (3.49s)
PASS
ok  	github.com/cybozu-go/moco/e2e	4.016s
```

File: docs/crd_mysql_cluster.md [hunk 1]
```diff
@@ -25,8 +25,7 @@ MySQLClusterSpec
...
 | `replicationSourceSecretName` | string                              | No       | `Secret` name which contains replication source info. Keys must appear in [Options].<br/> If this field is given, the `MySQLCluster` works as an intermediate primary (i.e., works as read-only replicas). |
-| `logRotationSchedule`         | string                              | No       | Schedule in Cron format for MySQL log rotation.                                                                                                                                                            |
-| `logRotationSecurityContext`  | [PodSecurityContext]                | No       | PodSecurityContext for MySQL log rotation Job's Pod.                                                                                                                                                       |
+| `logRotationSchedule`         | string                              | No       | Schedule in Cron format for MySQL log rotation.                                                                                                                                                            | |
```
Comment (masa213f) [inline docs/crd_mysql_cluster.md:28]:
```suggestion
| `logRotationSchedule`         | string                              | No       | Schedule in Cron format for MySQL log rotation.                                                                                                                                                            |
```

**Review by masa213f — APPROVED** [review]
Thank you!

**Pull request merged**